### PR TITLE
Throw error when pushing without wrapping into navigation page

### LINKED ISSUE #2013: [Bug] No error when try push outside NavigationPage
### Steps to Reproduce

1. Create a simple app, and make sure that `App.MainPage` is any `ContentPage`
2. Create a button and implement a clicked event that does `Navigation.PushAsync()`
3. Nothing happens

### Expected Behavior
1. See an exception or log that tells me that I need to have a `NavigationPage` as the root page.
2. When `App.MainPage` has a `NavigationPage` as the root page I want the breakpoint, inside click event handler, triggers (as you can see in the video that doesn't happen).
### Actual Behavior
Nothing happens

### Basic Information

- Version with issue: preview 6 (?)
- Last known good version: Na
- IDE: VS 2022
- Platform Target Frameworks: 
  - iOS:  
  - Android:  
  - UWP:   
- Android Support Library Version: 
- Nuget Packages: Na
- Affected Devices: Android

### Screenshots

https://user-images.githubusercontent.com/20712372/128791827-cc4f64e3-af88-4dfc-8544-ec79a1f63083.mp4

### Reproduction Link

[Navigation_Bug.zip](https://github.com/dotnet/maui/files/6958191/Navigation_Bug.zip)

### PR DESCRIPTION
**Pull request by kubaflo**
### Description of Change

Throw error when pushing without wrapping into navigation page

### Issues Fixed

Fixes https://github.com/dotnet/maui/issues/2013

### CODE REVIEW

**Comment by jfversluis** [comment]
As discussed on [this live stream](https://www.youtube.com/watch?v=g3UVNa5Bp48) this seems not the right place to do things, please let us know if anything is not clear on how to proceed to discuss further. Thank you!

**Review by jfversluis — CHANGES_REQUESTED** [review]
See above!

**Comment by kubaflo** [comment]
@jfversluis @PureWeen I've added a commit that moves this exception to the Window class and it seems to be working fine :)

File: src/Controls/src/Core/Window/Window.cs [hunk 1]
```diff
@@ -679,6 +679,11 @@ public NavigationImpl(Window owner)
 				_owner = owner;
 			}
 
+			protected override Task OnPushAsync(Page page, bool animated)
+			{
+				throw new InvalidOperationException("Page must be wrapped into a navigation page to perform navigation");
```
Comment (PureWeen) [inline src/Controls/src/Core/Window/Window.cs:684]: can we add this to all relevant overrides?

Notice how in XF we threw this on pretty much every override except the modal ones
https://github.com/xamarin/Xamarin.Forms/blob/5.0.0/Xamarin.Forms.Platform.iOS/Platform.cs#L143

**Review by PureWeen — CHANGES_REQUESTED** [review]

File: src/Controls/src/Core/Window/Window.cs (hunk 1 above)
Comment (kubaflo) [reply src/Controls/src/Core/Window/Window.cs]: Done :)

**Review by PureWeen — APPROVED** [review]

File: src/Controls/tests/Core.UnitTests/NavigationUnitTest.cs [hunk 2]
```diff
@@ -839,6 +839,29 @@ public async Task PopModalWithEmptyStackThrows()
...
 		}
 
+		[Fact]
+		public async Task InvalidOperationExceptionIsThrownWhenNavigatingOutsideNavigationPage()
+		{
+			var window = new TestWindow(new ContentPage());
+			var contentPage1 = new ContentPage();
+			var contentPage2 = new ContentPage();
+
+			Assert.ThrowsAsync<InvalidOperationException>(() => window.Navigation.PushAsync(contentPage1));
```
[the remark this answers is not in the document]
Comment (jfversluis) [reply src/Controls/tests/Core.UnitTests/NavigationUnitTest.cs:849]: Please add this case to your PR in #20506 so we can look at all of those in one place, thanks!

**Review by jfversluis — APPROVED** [review]

**Comment by jfversluis** [comment]
Thanks for another great contribution @kubaflo !

**Pull request merged**